internal/server/providers.go

```diff
@@ -75,6 +76,15 @@ func (a *API) CreateProvider(c *gin.Context, r *api.CreateProviderRequest) (*api
 	}
 	provider.Kind = kind
 
+	// If name is not provided, generate based on provider kind
+	if provider.Name == "" {
+		randomString, err := generate.CryptoRandom(6, generate.CharsetAlphaNumeric)
+		if err != nil {
+			return nil, errors.New("Error while generating name for provider")
+		}
+		provider.Name = r.Kind + "-" + randomString
```

Comment:
I think there was some discussion about not adding the random suffix, and instead using the kind as the name without any suffix.
Do you think that could work?

If someone adds multiple providers of the same kind they would be forced to pick a name, but I think we expect almost everyone to use only a single provider of any kind, so that should be very rare.

Reply:
Yep! Just uploaded a video of the new behavior. It will check how of that kind there is and append a number